open-craft/blockstore
# LX-53 REST API improvements

Provides several improvements to the REST API, detailed below.

**JIRA tickets**:  LX-53, SE-281

**Testing instructions**:

- [x] Add expand field to /bundles/ and /bundle_versions/ to inline the files list:  b25d9a6 
   * Visit http://localhost:8888/api/v1/bundle/ and view a bundle version.
   * Visit http://localhost:8888/api/v1/bundle_versions/ and view a bundle version.
   * Append `?expand=files` to the urls above, and note that the `files` hyperlink is converted to an array of files data, which matches what was shown by following the `files` hyperlink.
   * Appending `?expand=files&format=json` also works.
   * Appending `?expand=files` to the `/api/v1/bundle/` and `/api/v1/bundle_versions/` list views has no effect (previous to f47b553, it threw an error).
- [x] Add filters to the endpoints: c47b482
   * list Bundles associated with a Collection, e.g. 
      http://localhost:8888/api/v1/bundles/?collection__uuid=4ffa83df9b93499e91c29f5a7ad9
   * list BundleVersions for a Bundle, e.g.
      http://localhost:8888/api/v1/bundle_versions/?bundle__uuid=33566cb9-e607-425b-8cd1-db0ad4ab97d7
   * list BundleVersions for a Collection, e.g.
      http://localhost:8888/api/v1/bundle_versions?bundle_collection__uuid=4ffa83df-9b93-499e-91c29f5a7ad9
- [x] Add PUT method to `BundleVersionViewSet` to allow changing `change_description`: 023d251
   * Visit http://localhost:8888/api/v1/bundle_versions/ and view a bundle version.
   * Update the `change_description` field and submit.
- [x] Allow posting multiple files at once: 3c93aee
   * This change cannot be tested from the DRF GUI
   * See the [`test_create_multiple_files`](https://github.com/open-craft/blockstore/commit/3c93aee20a9dead5c5036b816861f1da0f28c203#diff-3959f0e2e005ddd8ad570fb6febdea94R431) unit test for details on how this feature works.
- [x] Verify the exact output in view tests: 6fdae38 

**Reviewers**
